### PR TITLE
[ST] Refactor CustomCa STs

### PR DESCRIPTION
**Pull request by jankalinic**
### Type of change

- Refactoring

### Description

In this PR there are a few improvements to Custom Certificate Authority STs.
Additionally, it introduces a newly added / split tests specifically designed to validate manual certificate renewal processes. These tests are designed for verifying reliability of certificate renewals.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=ocp --cluster-version=4.14 --install-type=bundle --testcase=CustomCaST

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:464]: I think this helper method deserves a proper description with steps (Javadoc)

Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:477]: Thanks. Added

**Review by im-konge — APPROVED** [review]

**Comment by jankalinic** [comment]
Tests on azp buildId=155378 finished with one error -> testKafkaDoesNotRollsWhenTopicIsUnderReplicated

**Comment by scholzj** [comment]
This is not ready for merge because it does not have the required number of reviews from maintainers.

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java
Comment (jankalinic) [reply systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:71]: Sure, will add some docs with steps. Thanks

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java
Comment (jankalinic) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaResource.java:179]: Considering it should follow other resources, yes, good point, thank you

**Comment by jankalinic** [comment]
@strimzi-ci run tests --cluster-type=minikube --cluster-version=1.27 --install-type=bundle --testcase=CustomCaST

**Review by Frawless — APPROVED** [review]

**Comment by im-konge** [comment]
I accidentally started kraft-regression again, so I cancelled it -> the last successful results are https://dev.azure.com/cncf/strimzi/_build/results?buildId=156184&view=results

**Pull request merged**
